File: examples/prpc_client.py
import a0
import time


def callback(pkt, done):
    print(f'Progress info: {pkt.payload.decode("utf-8")}')
    if done:
        print("Completed")


print("Waiting 1s for responses")
client = a0.PrpcClient("topic")
client.connect("client request", callback)
time.sleep(1)
print("Done!")

File: examples/prpc_client.go
package main

import (
	"fmt"
	a0 "github.com/alephzero/go"
	"log"
	"time"
)

func check(err error) {
	if err != nil {
		log.Panicf("err: %v", err)
	}
}

func main() {
	topic, err := a0.FileOpen("alephzero/example.prpc.a0", nil)
	check(err)
	defer topic.Close()

	client, err := a0.NewPrpcClient(topic)
	check(err)
	defer client.Close()

	req := a0.NewPacket(nil, []byte("client request"))

	fmt.Println("Waiting 1s for responses")

	check(client.Connect(req, func(pkt a0.Packet, done bool) {
		fmt.Printf("Progress info: %v\n", string(pkt.Payload))
		if done {
			fmt.Println("Completed")
		}
	}))

	time.Sleep(time.Second)

	fmt.Println("Done!")
}
